django/db/backends/base/base.py

```diff
@@ -738,6 +738,20 @@ def on_commit(self, func):
             # immediately.
             func()
 
+    def on_commit_robust(self, func):
+        if not callable(func):
+            raise TypeError("on_commit()'s callback must be a callable.")
+        if self.in_atomic_block:
+            # Transaction in progress; save for execution on commit.
+            self.run_on_commit.append((set(self.savepoint_ids), func))
+        else:
+            # No transaction in progress and in autocommit mode; execute
+            # immediately.
+            try:
+                func()
+            except Exception as e:
+                print(f"{e} occurred while calling {func.__name__}")
```

Comment:
Use a similar logging to what we do for robust signals?
See `django.dispatch.dispatcher.Signal.send_robust` and
```
            try:
                response = receiver(signal=self, sender=sender, **named)
            except Exception as err:
                logger.error(
                    "Error calling %s in Signal.send_robust() (%s)",
                    receiver.__qualname__,
                    err,
                    exc_info=err,
                )
```
Something like
```suggestion
                logger.error("Error calling {func.__qualname__} on_commit() ({e}).", exc_info=True)
```

Reply:
I will make the appropriate changes.